Scripts/RemediationScripts/Remediate-EnableTransparentDataEncryption.ps1

```diff
@@ -0,0 +1,653 @@
+    }
+
+    Write-Host "Fetching all SQL Server database details from $($FilePath)" -ForegroundColor $([Constants]::MessageType.Info)
+
+    $sqlServerDetails = Import-Csv -LiteralPath $FilePath
+    $validSqlServerDetails = $sqlServerDetails | Where-Object { ![String]::IsNullOrWhiteSpace($_.ResourceGroupName) -and ![String]::IsNullOrWhiteSpace($_.ServerName) -and ![String]::IsNullOrWhiteSpace($_.DatabaseName)}
+
+    $totalSqlServerDatabases = $($validSqlServerDetails.Count)
+
+    if ($totalSqlServerDatabases -eq 0)
+    {
+        Write-Host "No SQL Server database found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        break
+    }
+
+    Write-Host "Found $($totalSqlServerDatabases) SQL Server database(s)." -ForegroundColor $([Constants]::MessageType.Update)
+    
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\EnableTDEForSQLServers"
+
+    if (-not (Test-Path -Path $backupFolderPath))
+    {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+ 
+    Write-Host "Transparent Data Encryption (TDE) will be disabled for SQL Server databases." -ForegroundColor $([Constants]::MessageType.Warning)
+
+    
+    if (-not $Force)
+    {
+        Write-Host "Do you want to disable Transparent Data Encryption (TDE) for all SQL Server databases? " -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+            
+        $userInput = Read-Host -Prompt "(Y|N)"
+
+        if($userInput -ne "Y")
+        {
+            Write-Host "Transparent Data Encryption (TDE) will not be disabled for SQL Server databases. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+            break
+        }
+    }
+    else
+    {
+        Write-Host "'Force' flag is provided. Transparent Data Encryption (TDE) will be disabled for SQL Server databases." -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+    }
+
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 3 of 3] Disabling Transparent Data Encryption (TDE) for SQL Server databases..." -ForegroundColor $([Constants]::MessageType.Warning)
+
+    # Includes SQL Server databases, to which, previously made changes were successfully rolled back.
+    $sqlServersRolledBack = @()
+
+    # Includes SQL Server databases that were skipped during roll back. There were errors rolling back the changes made previously.
+    $sqlServersSkipped = @()
+
+    $validSqlServerDetails | ForEach-Object {
+        $databaseDetail = $_
+        $ServerName = $_.ServerName
+        $DatabaseName = $_.DatabaseName
+        $ResourceGroupName = $_.ResourceGroupName
+        try
+        {
+            $tdeStatus = $(Set-AzSqlDatabaseTransparentDataEncryption -ResourceGroupName $ResourceGroupName -ServerName $ServerName -DatabaseName $DatabaseName -State 'Disabled').State
```

Comment:
While I see merit in doing this, can we remediate/roll back all SQL Databases in a SQL Server or None. Ideally, users won't selective enable/disable TDE for a set of databases in the server.